main.py

```diff
@@ -110,16 +124,100 @@ def main(
             paths[index] = queue.path
 
         if not paths[index] is None:
-            if index == 1:
-                dc.clear_query_results(path=paths[index], step=3600)
-                continue
+            try:
+                step = queries[index].kwargs["params"]["step"]
+            except KeyError:
+                step = 60
 
-            dc.clear_query_results(path=paths[index], step=60)
+            dc.clear_query_results(path=paths[index], step=step)
 
     end = dt.now()
     logger.info("Cleaning data lastet: %s seconds.", (end - start))
 
+    logger.info("Starting to analyze data.")
+    start = dt.now()
+
+    for path in paths:
+        if path is None:
+            continue
+        analyzer.get_mean_duration_per_alertname(path=path)
+
+    end = dt.now()
+    logger.info("Analyzing data lastet: %s seconds.", (end - start))
+
+    for index_path, path in enumerate(paths[0:1]):
+        filtered_data = analyzer.group_alert_timeseries_per_cluster(path=path)
+        start_tt = float(queries[index_path].global_start)
+        end_tt = float(queries[index_path].global_end)
+        correlated_data = analyzer.correlate_data(
+            path=path,
+            result=filtered_data,
+            gap=60,
+            cores=cores,
+            start_tt=start_tt,
+            end_tt=end_tt,
+        )
+        analyzer.create_alert_corrrelation_list(
+            path=path, alerts=correlated_data["alert_index"], matrix=correlated_data["corrcoef_matrix"]
+        )
+
+    return paths
+
+
+def main(
+    api_endpoint: str = None,
+    cert: str = None,
+    timeout: int = None,
+    kwargs: dict = None,
+    directory_path: str = None,
+    threshold: int = None,
+    delay: float = None,
+    cores: int = None,
+    max_long_term_storage: str = None,
+    prometheus_port: int = None,
+    **kkwargs  # additional unused keyword arguments for logging purposes
+):
+    logger.debug("Starting main function with config: %s", CONFIG)
+
+    if kwargs is None:
+        kwargs = {}
+
+    e = Exporter(prometheus_port=prometheus_port, paths=[])
+    exporter_thread = Thread(target=e.start_server)
+    exporter_thread.start()
+
+    to_be_removed_directories = []
+
+    while True:
+        paths = do_analysis(
+            api_endpoint=api_endpoint,
+            cert=cert,
+            timeout=timeout,
+            kwargs=kwargs,
+            directory_path=directory_path,
+            threshold=threshold,
+            delay=delay,
+            cores=cores,
+            max_long_term_storage=max_long_term_storage,
+        )
+
+        e.paths = paths[0:1]
+        e.increase_alertmagnet_analyzing_count()
+        e.update_metrics()
+
+        time.sleep(60 * 60 * 24)
```

Comment:
this needs to be configureable to have the possibility to run it more often or delay it further.